pkg/controller/database/rdsinstance/rdsinstance.go

```diff
@@ -277,12 +277,18 @@ func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalUpdate{}, err
 	}
 	if changed {
-		conn = managed.ConnectionDetails{
-			xpv1.ResourceCredentialsSecretPasswordKey: []byte(pwd),
-		}
+		conn = managed.ConnectionDetails{xpv1.ResourceCredentialsSecretPasswordKey: []byte(pwd)}
 		modify.MasterUserPassword = aws.String(pwd)
 	}
 
+	if cr.Spec.ForProvider.MasterUsername != nil {
+		if conn != nil {
+			conn[xpv1.ResourceCredentialsSecretUserKey] = []byte(aws.ToString(cr.Spec.ForProvider.MasterUsername))
+		} else {
+			conn = managed.ConnectionDetails{xpv1.ResourceCredentialsSecretUserKey: []byte(aws.ToString(cr.Spec.ForProvider.MasterUsername))}
+		}
+	}
+
```

Comment:
I think that can be simplied:

```suggestion
	conn := managed.ConnectionDetails{}
	
	if changed {
		conn[xpv1.ResourceCredentialsSecretPasswordKey] = []byte(pwd)
	}

	if cr.Spec.ForProvider.MasterUsername != nil {
		conn[xpv1.ResourceCredentialsSecretUserKey] = []byte(aws.ToString(cr.Spec.ForProvider.MasterUsername))
	}

```

Reply:
Updated, the only reason I didn't do this originally is because I wasn't sure if `managed.ConnectionDetails{}` needed to be `nil` not `empty`.
But then while implementing it I realised it should never be empty because `Spec.ForProvider.MasterUsername` should always be set.
Thanks!